elide-async/src/test/java/com/yahoo/elide/async/integration/tests/AsyncIntegrationTest.java

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.integration.tests;
+
+import static com.yahoo.elide.Elide.JSONAPI_CONTENT_TYPE;
+import static com.yahoo.elide.contrib.testhelpers.jsonapi.JsonApiDSL.attr;
+import static com.yahoo.elide.contrib.testhelpers.jsonapi.JsonApiDSL.attributes;
+import static com.yahoo.elide.contrib.testhelpers.jsonapi.JsonApiDSL.data;
+import static com.yahoo.elide.contrib.testhelpers.jsonapi.JsonApiDSL.id;
+import static com.yahoo.elide.contrib.testhelpers.jsonapi.JsonApiDSL.resource;
+import static com.yahoo.elide.contrib.testhelpers.jsonapi.JsonApiDSL.type;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+
+import javax.persistence.Persistence;
+import javax.ws.rs.core.MediaType;
+
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import com.yahoo.elide.async.integration.framework.AsyncDataStoreTestHarness;
+import com.yahoo.elide.async.integration.framework.AsyncIntegrationTestApplicationResourceConfig;
+import com.yahoo.elide.core.HttpStatus;
+import com.yahoo.elide.core.datastore.test.DataStoreTestHarness;
+import com.yahoo.elide.initialization.IntegrationTest;
+import com.yahoo.elide.resources.JsonApiEndpoint;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class AsyncIntegrationTest extends IntegrationTest{
+
+    public AsyncIntegrationTest() {
+        super(AsyncIntegrationTestApplicationResourceConfig.class, JsonApiEndpoint.class.getPackage().getName());
+    }
+
+    @Override
+    protected DataStoreTestHarness createHarness() {
+        return new AsyncDataStoreTestHarness(Persistence.createEntityManagerFactory("asyncStore"));
+    }
+
+    /**
+     * This test demonstrates an example post request using JSON-API for a JSONAPI query.
+     */
+    @Test
+    @Order(1)
```

Comment:
No test orders please.  Elide used to use these - and these decayed over time to something very unmaintainable.  Each tests should create whatever test data it needs - and then clean it up when done.  Each test should stand on its own.